private/transducer-sorting.rkt

```diff
@@ -74,6 +62,11 @@
   #f)
 
 (define (sorting [comparator real<=>] #:key [key-function values])
+  ;; TODO(https://github.com/jackfirth/): handle key function more efficiently
+  ;;   by caching keys, and avoid computing them altogether when possible (e.g.
+  ;;   singleton sequence).
+  (define keyed-comparator (comparator-map real<=> key-function))
```

Comment:
Shouldn't this be:

`(define keyed-comparator (comparator-map comparator key-function))`

Reply:
Oops. Definitely it should be that. Great catch, thank you!